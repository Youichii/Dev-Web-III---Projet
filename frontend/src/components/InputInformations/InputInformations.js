import PropTypes from 'prop-types'

const InputInformations = ({ className_div, type, id_input, maxLenght, id_span }) => {
    return (
        <div className={className_div}>
            <input data-testid="id_input_info" type={type} id={id_input} required maxLength={maxLenght} /> 
            <span className="message_erreur" id={id_span}></span>
        </div>
    )
}

InputInformations.defaultProps = {
    type: 'text'
}

InputInformations.propTypes = {
    informations : PropTypes.string,
    type_couleur : PropTypes.string
}

export default InputInformations
