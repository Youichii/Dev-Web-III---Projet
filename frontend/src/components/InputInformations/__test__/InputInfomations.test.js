import React from 'react';
import ReactDOM from 'react-dom';
import InputInformations from '../InputInformations';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer  from "react-test-renderer";

afterEach(cleanup);
it("renders without crashing InputInformation", () => {
    const div = document.createElement("div");
    ReactDOM.render(<InputInformations></InputInformations>, div)
})

it("render id input correctly InputInformation", () => {
    const {getByTestId} = render(<InputInformations className_div="i_champ_adresse" id_input="text_user" id_span="erreur_mail" maxLenght="100"></InputInformations>);
    expect(getByTestId('id_input_info')).toHaveAttribute('id', "text_user");
})

it("render id input correctly InputInformation", () => {
    const {getByTestId} = render(<InputInformations className_div="i_champ_adresse" id_input="text_user" id_span="erreur_mail" maxLenght="100"></InputInformations>);
    expect(getByTestId('id_input_info')).toHaveAttribute('type', "text");
})

it("matches snapshot 1 InputInformation", () => {
    const tree = renderer.create(<InputInformations className_div="i_champ_adresse" id_input="text_user" id_span="erreur_mail" maxLenght="100"></InputInformations>).toJSON();
    expect(tree).toMatchSnapshot(); 
}) 

it("matches snapshot 2 InputInformation", () => {
    const tree = renderer.create(<InputInformations className_div="i_champ_adresse" id_input="text_user" id_span="erreur_mail" maxLenght="100"></InputInformations>).toJSON();
    expect(tree).toMatchSnapshot();
}) 