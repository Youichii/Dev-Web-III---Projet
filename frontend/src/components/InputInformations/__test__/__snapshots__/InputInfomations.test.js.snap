// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 InputInformation 1`] = `
<div
  className="i_champ_adresse"
>
  <input
    data-testid="id_input_info"
    id="text_user"
    maxLength="100"
    required={true}
    type="text"
  />
  <span
    className="message_erreur"
    id="erreur_mail"
  />
</div>
`;

exports[`matches snapshot 2 InputInformation 1`] = `
<div
  className="i_champ_adresse"
>
  <input
    data-testid="id_input_info"
    id="text_user"
    maxLength="100"
    required={true}
    type="text"
  />
  <span
    className="message_erreur"
    id="erreur_mail"
  />
</div>
`;
